cli/azd/pkg/tools/maven/maven.go

```diff
@@ -237,6 +238,48 @@ func (cli *Cli) GetProperty(ctx context.Context, propertyPath string, projectPat
 	return result, nil
 }
 
+func (cli *Cli) EffectivePom(ctx context.Context, pomPath string) (string, error) {
+	mvnCmd, err := cli.mvnCmd()
+	if err != nil {
+		return "", err
+	}
+	pomDir := filepath.Dir(pomPath)
+	runArgs := exec.NewRunArgs(mvnCmd, "help:effective-pom", "-f", pomPath).WithCwd(pomDir)
+	result, err := cli.commandRunner.Run(ctx, runArgs)
+	if err != nil {
+		return "", fmt.Errorf("failed to run mvn help:effective-pom for pom file: %s. error = %w", pomPath, err)
+	}
+	return getEffectivePomFromConsoleOutput(result.Stdout)
+}
+
+func getEffectivePomFromConsoleOutput(consoleOutput string) (string, error) {
+	var builder strings.Builder
+	scanner := bufio.NewScanner(strings.NewReader(consoleOutput))
+	inProject := false
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(strings.TrimSpace(line), "<project") {
+			inProject = true
+		} else if strings.HasPrefix(strings.TrimSpace(line), "</project>") {
+			builder.WriteString(line)
+			break
+		}
+		if inProject {
+			if strings.TrimSpace(line) != "" {
+				builder.WriteString(line)
+			}
+		}
+	}
```

Comment:
a small suggestion here to avoid string allocations and streamline the code

```suggestion
	inProject := false
	projectStart := regexp.MustCompile(`\s+<project`)
	projectEnd := regexp.MustCompile(`\s+</project>`)

	for scanner.Scan() {
		line := scanner.Text()
		if projectStart.MatchString(line) {
			inProject = true
		} else if projectEnd.MatchString(line) {
			builder.WriteString(line)
			break
		}
		if inProject {
		  builder.WriteString(line)
		}
	}
```

Reply:
FYI:
1. I'll accept the changes by clicking `commit suggestion`.
2. I'll fix the indent problem in line 271 in another commit.
3. To test the changes, make sure it won't bring new bug, I'll recover `pom_test.go`.

Reply:
FYI:

> To test the changes, make sure it won't bring new bug, I'll recover pom_test.go.

Test failed.

Here is the screenshot when I debug:

> ![image](https://github.com/user-attachments/assets/6843365a-6daa-4214-adc4-6e403db532eb)

1. I'll fix the bug in another commit.
2. It shows that `pom_test.go` is useful, I'll keep it.

Reply:
> I'll fix the bug in another commit.

Done in this commit: https://github.com/Azure/azure-dev/pull/4710/commits/90e806f32ec9c4a74864b29eae8bedb70763b4b9

Reply:
@weikanglim , @saragluna 

FYI: Here is the current output when `mvn` command not exist:

> ![image](https://github.com/user-attachments/assets/eae79eb2-592c-43f5-8339-40eb6861414d)